x-pack/plugins/ml/public/application/components/model_snapshots/edit_model_snapshot_flyout/edit_model_snapshot_flyout.tsx

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiFlyout,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiSpacer,
+  EuiTextArea,
+  EuiFormRow,
+  EuiSwitch,
+  EuiConfirmModal,
+  EuiOverlayMask,
+  EuiCallOut,
+} from '@elastic/eui';
+
+import {
+  ModelSnapshot,
+  CombinedJobWithStats,
+} from '../../../../../common/types/anomaly_detection_jobs';
+import { ml } from '../../../services/ml_api_service';
+import { useNotifications } from '../../../contexts/kibana';
+
+interface Props {
+  snapshot: ModelSnapshot;
+  job: CombinedJobWithStats;
+  closeFlyout(reload: boolean): void;
+}
+
+export const EditModelSnapshotFlyout: FC<Props> = ({ snapshot, job, closeFlyout }) => {
+  const { toasts } = useNotifications();
+  const [description, setDescription] = useState(snapshot.description);
+  const [retain, setRetain] = useState(snapshot.retain);
+  const [deleteModalVisible, setDeleteModalVisible] = useState(false);
+  const isCurrentSnapshot = snapshot.snapshot_id === job.model_snapshot_id;
+
+  async function updateSnapshot() {
+    try {
+      await ml.updateModelSnapshot(snapshot.job_id, snapshot.snapshot_id, {
+        description,
+        retain,
+      });
+      closeWithReload();
+    } catch (error) {
+      toasts.addError(new Error(error.body.message), {
+        title: i18n.translate('xpack.ml.editModelSnapshotFlyout.saveErrorTitle', {
+          defaultMessage: 'Model snapshot update failed',
+        }),
+      });
+    }
+  }
+
+  async function deleteSnapshot() {
+    try {
+      await ml.deleteModelSnapshot(snapshot.job_id, snapshot.snapshot_id);
+      hideDeleteModal();
+      closeWithReload();
+    } catch (error) {
+      toasts.addError(new Error(error.body.message), {
+        title: i18n.translate('xpack.ml.editModelSnapshotFlyout.deleteErrorTitle', {
+          defaultMessage: 'Model snapshot deletion failed',
+        }),
+      });
+    }
+  }
```

Comment:
You will get a new instance on each render. Consider wrapping it with `useCallback`

Reply:
Changed in ac22c8ce86162fd9d4db4687850de4ae33873f4b